src/System.Device.Gpio/System/Device/Gpio/GpioController.Linux.cs

```diff
@@ -28,19 +28,37 @@ public GpioController(PinNumberingScheme numberingScheme)
         /// Attempt to get the best applicable driver for the board the program is executing on.
         /// </summary>
         /// <returns>A driver that works with the board the program is executing on.</returns>
-        private static GpioDriver GetBestDriverForBoard()
+        public static GpioDriver GetBestDriverForBoard()
```

Comment:
Why do we want to change this to public? The main idea here is that you either want to get the driver automatic selection done for you which the default constructor is already doing, or you want to pass it in yourself, so I don't see why we would ever want to make this API public

Reply:
what about making driver which extends/augments driver?

Reply:
Or (as in the test) check whether the current driver is the suggested one?

Reply:
For both of those cases, I believe we should have user-defined logic for calculating that. This is an API which by definition will have breaking changes for every new board we add, so we definitely don't want to make it public and take breaking change after breaking change. Keeping it private and changing the logic is ok, and for augmenting driver we should define a different function, and for tests we might consider using InternalsVisibleTo, which is ugly, but better than exposing this API.